### PR TITLE
feat: ai schema framework

### PR DESCRIPTION
**Pull request by dspo**
#### Which issue(s) this PR fixes:

- Fixes #your-issue_number
- [Erda Cloud Issue Link](paste your link here)

#### Specified Reviewers:

/assign @your-reviewer

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English | add ai function framework             |
| 🇨🇳 中文    | 添加 ai function 框架 |

#### Need cherry-pick to release versions?

Add comment like `/cherry-pick release/1.0` when this PR is merged.

> For details on the cherry pick process, see the [cherry pick requests](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md#how-to-cherry-pick-a-merged-pr) section under [CONTRIBUTING.md](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: internal/pkg/ai-functions/sdk/client.go [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+}
+
+func (c *Client) CreateCompletion(ctx context.Context, req *CreateCompletionOptions) (*ChatCompletions, error) {
+	req.Stream = false
+	var buf bytes.Buffer
+	if err := json.NewEncoder(&buf).Encode(req); err != nil {
+		return nil, errors.Wrap(err, "failed to Encode CreateCompletionOptions")
+	}
+	request, err := http.NewRequest(http.MethodPost, c.URLV1ChatCompletion(), &buf)
+	request.Header.Set("Authorization", "Bearer "+"e78c1fe49d704fda978041cb21770282") // todo: ak
```
Comment (sfwn) [inline internal/pkg/ai-functions/sdk/client.go:67]: what's this

**Review by sfwn — CHANGES_REQUESTED** [review]
Remove bin file

File: .erda/ai-proxy/migrations/ai-proxy/20230425-filter-audit.sql [hunk 2]
```diff
@@ -24,6 +24,10 @@ CREATE TABLE `ai_proxy_filter_audit`
     `operation_id`          VARCHAR(128) NOT NULL COMMENT '调用的接口名称, HTTP Method + Path',
     `prompt`                MEDIUMTEXT   NOT NULL COMMENT '提示语',
     `completion`            LONGTEXT     NOT NULL COMMENT 'AI 回复多个 choices 中的一个',
+    `req_func_call_name`    VARCHAR(128) NOT NULL COMMENT 'function_call name in request message',
```
[the remark this answers is not in the document]
Comment (dspo) [reply .erda/ai-proxy/migrations/ai-proxy/20230425-filter-audit.sql:27]: 留给你啦

File: bundle/testcase.go [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+	log "github.com/sirupsen/logrus"
+
+	"github.com/erda-project/erda/apistructs"
+	"github.com/erda-project/erda/bundle/apierrors"
+	"github.com/erda-project/erda/pkg/http/httputil"
+)
+
+func (b *Bundle) CreateTestCase(req apistructs.TestCaseCreateRequest) ([]byte, error) {
+	resp, err := b.hc.Post("localhost:9095").Path("/api/testcases").
```
[the remark this answers is not in the document]
Comment (dspo) [reply bundle/testcase.go:28]: 请实现 gRPC 接口或本地调用

File: cmd/erda-server/bootstrap.yaml [hunk 4]
```diff
@@ -618,3 +618,5 @@ grpc-client@erda.core.monitor.settings:
   block: false
 erda.core.monitor.settings-client:
 service-profile-overview:
+erda.app.ai-function:
+  openAIOpenapi: ${OPENAI_OPENAPI:http://ai-proxy:8081}
```
[the remark this answers is not in the document]
Comment (dspo) [reply cmd/erda-server/bootstrap.yaml:622]: rename to `openaiAddr`

File: internal/pkg/ai-functions/sdk/client.go (hunk 1 above)
Comment (dspo) [reply internal/pkg/ai-functions/sdk/client.go:67]:
```go
	// 在 request option 中添加认证信息: 以某组织下某用户身份调用 ai-proxy,
	// ai-proxy 中的 filter erda-auth 会回调 erda.cloud 的 openai, 检查该企业和用户是否有权使用 AI 能力
	ros := append(f.RequestOptions(), func(r *http.Request) {
		r.Header.Set("X-Ai-Proxy-Source", "erda.cloud") // todo: hard code source
		r.Header.Set("X-Ai-Proxy-Org-Id", base64.StdEncoding.EncodeToString([]byte(apis.GetOrgID(ctx))))
		r.Header.Set("X-Ai-Proxy-User-Id", base64.StdEncoding.EncodeToString([]byte(apis.GetUserID(ctx))))
	})
```

**Review by sfwn — APPROVED** [review]

**Pull request merged**